src/core/effutil.ts

```diff
@@ -0,0 +1,37 @@
+import {Eff} from './eff';
+import {Run} from './run';
+
+class Deferred<T> {
+    promise: Promise<T> = new Promise<T>((resolve, reject) => {
+        this.resolve = resolve;
+        this.reject = reject;
+    });
+    resolve: (t: T) => void;
+    reject: (e: Error) => void;
+}
+
+export class EffUtil {
+    static scheduledOnce<T>(block: () => T, delay: number): Eff<{}, T> {
```

Comment:
I think this (and the other utilities) should be "for all effects", so this should look more like `scheduledOnce<F, T>(block: () => T, delay: number): Eff<F, T>`.